### DISCUSSION TITLE
Actions/checkout@v3 can't authenticate on: pull_request

### DISCUSSION THREAD

**Discussion by levid0s** [community/community#26658] [Actions]
<p>I’m working on a pipeline that will perform some checks on the source code before auto-approving a merge request. However, I’m having issues checking out the source code.</p>
<p>I tried the <strong>pull_request</strong> and <strong>pull_request_target</strong> events.</p>
<p>My pipeline looks like this:</p>
<pre><code class="lang-auto">name: Auto Approval Pipeline
on:  pull_request
jobs:
  pr-auto-approval:
    runs-on: ubuntu-latest
    permissions:
      pull-requests: write
    steps:
    - uses: actions/checkout@v3
      with:
      # ref: ${{ github.event.pull_request.head.sha }} # with pull_request_target I had to add this line to checkout the right branch.
        token: ${{ secrets.GITHUB_TOKEN }} # tried with and without this line
    - name: Validate json
      id: validate-json
      run: |
        jq . file.json
        [[ "$?" != "0" ]] &amp;&amp; exit 1
        echo "INFO: JSON valid."
    - name: auto-approve-action
      uses: hmarr/auto-approve-action@v2
      with:
        github-token: ${{ secrets.GITHUB_TOKEN }} 
</code></pre>
<p>Error:</p>
<pre><code class="lang-auto">  /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +a02c69be65b1693fe2119d9cce38ee240f68f24c:refs/remotes/pull/9/merge
  remote: Repository not found.
  Error: fatal: repository 'https://github.com/user/myproject/' not found
</code></pre>
<p>On a regular pipeline I can check out the code all fine using:</p>
<pre><code class="lang-auto">name: Feature Branch Pipeline
on: push
jobs:
  process-json:
    runs-on: ubuntu-latest
    name: Process JSON
    steps:
    - uses: actions/checkout@v3
    - name: First script
      run: |
        ls -lRh .
        jq . file.json
</code></pre>
<p>Any ideas appreciated.</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="KbLev" data-post="1" data-topic="251500">
<div class="title">
<div class="quote-controls"></div>
 KbLev:</div>
<blockquote>
<pre><code class="lang-yaml">    permissions:
      pull-requests: write
</code></pre>
</blockquote>
</aside>
<p>Here’s your problem. Without read access to <code>contents</code> the job can’t access the repository contents.</p>

**Comment by levid0s** [comment]
<p>Thank you. Adding <code>permissions: contents: read</code> fixed my issue.</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions">
  <header class="source">
      <img src="https://docs.github.com/assets/cb-803/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions" target="_blank" rel="noopener nofollow ugc">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/37480688/181099094-eab7ccda-5142-45f9-a6cc-d41fa3f41d29.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions" target="_blank" rel="noopener nofollow ugc">Workflow syntax for GitHub Actions - GitHub Docs</a></h3>

  <p>A workflow is a configurable automated process made up of one or more jobs. You must create a YAML file to define your workflow configuration.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
